rootston/seat.c

```diff
@@ -800,6 +804,37 @@ void roots_seat_set_focus(struct roots_seat *seat, struct roots_view *view) {
 	}
 }
 
+/**
+ * Focus semantics of layer surfaces are somewhat detached from the normal focus
+ * flow. For layers above the shell layer, for example, you cannot unfocus them.
+ * You also cannot alt-tab between layer surfaces and shell surfaces.
+ */
+void roots_seat_set_focus_layer(struct roots_seat *seat,
+		struct wlr_layer_surface *layer) {
+	struct wlr_keyboard *keyboard = wlr_seat_get_keyboard(seat->seat);
+	if (!layer) {
+		seat->focused_layer = NULL;
+		return;
+	}
+	if (seat->has_focus) {
```

Comment:
What happens if there's a fullscreen view? What happens if the view is focused by another client?

Calling `roots_seat_set_focus(seat, NULL)` could work better.

Reply:
So what if there's a fullscreen view? You can't click layers beneath the shell layer if there is. Layers above the shell layer behave as expected.